research-software-directory/RSD-as-a-service
# Helmholtz login improvements

# Helmholtz login improvements

Fixes https://github.com/research-software-directory/Helmholtz/issues/16

Changes proposed in this pull request:

* as we realized that also social providers and many international institutes are listed for Helmholtz AAI login we want to make sure that only Helmholtz Centres can login
* by default only Helmholtz centres listed in https://hifis.net/doc/helmholtz-aai/list-of-vos/ can login
* with the environment variable `HELMHOLTZAAI_ALLOW_EXTERNAL_USERS` set to `true` all logins via Helmholtz AAI are allowed, including non-Helmholtz institutes and social IdPs

How to test:

* set `HELMHOLTZAAI_ALLOW_EXTERNAL_USERS` to `true`, go to frontend website `localhost/login`, click on Helmholtz AAI logo and log in with Google or ORCID
* set `HELMHOLTZAAI_ALLOW_EXTERNAL_USERS` to `false` or comment out, login again with Google or ORCID and it should not work

Unfortunately I did not find a way to send the frontend the reason why the login did not work. Because it may did not go wrong, but the user is just not allowed to login since the RSD is configured this way.

PR Checklist:
